### PR TITLE
Makefile: avoid rebuilding binaries

### PR DESCRIPTION
**Pull request by kolyshkin**
#### What type of PR is this?

/kind cleanup
/kind regression

#### What this PR does / why we need it:

_(this is a continuation of #684; sorry for the mess)_

It is a common scenario in which someone runs

	make
	sudo make install

Problem is, with the current Makefile, which has install-crictl depend
on a phony target crictl (and same for critest), it leads to making
crictl (and critest) targets twice. It is OK per se, since go is very
effective in caching dependencies and build artefacts, but it can be
problematic with sudo in case go is installed in some non-standard
path and/or requires some environment variables that sudo skips
(unless -E is used). As a result, binaries can be rebuilt again with
the wrong version of go, or some wrong environment, or maybe go binary
can't be found at all due to changed $PATH.

Anyway, to fix all this, let's create real (rather than phony) targets,
and have install depend on it.

At the same time, it would be good to have make always try to rebuild
the binaries, since it's not aware of their dependencies (but go does).
So, have the phony targets `crictl` and `critest` which run `make -B`
to force rebuild.

In other words, `make` (or `make binaries`, or `make crictl`) will now
always rebuild the binaries (same as before), but `make install` will
only try to build binaries if they are absent.

While at it:
 - put the built binaries into BUILD_BIN_DIR
 - consolidate and simplify some targets
 - remove non-existent targets from .PHONY

Signed-off-by: Kir Kolyshkin <kolyshkin@gmail.com>

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by kolyshkin** [comment]
This is still messed up. Ideally, I'd love to have `make` rebuild the binaries, but `make install` to use the prebuilt ones, if available, and rebuild them if not. Not sure how to implement it though. Need to think.

**Comment by kolyshkin** [comment]
Rebased to remove the stale flag.

Is there anything that I need to address?

**Comment by kolyshkin** [comment]
@saschagrunert PTAL

**Review by saschagrunert — APPROVED** [review]
/lgtm
